### DISCUSSION TITLE
Change color of focused screen Bar

### DISCUSSION THREAD

**Discussion by asyncmind0** [qtile/qtile#2610] [Q&A]
Hi I'm looking for a way to hint which screen has focus currently

I have a quad monitor setup and figuring out which screen has focus gets tricky 

I want the bar on the active screen to have  a different background

Did not find any focus related code in Bar

**Comment by m-col** [comment]
Perhaps you could use the `focus_change` hook to set the bar 
background, for example (untested):

```py
from libqtile import hook, qtile

@hook.subscribe.focus_change
def _():
    for screen in qtile.screens:
        if screen is qtile.current_screen:
            screen.bottom.background = focussed_colour
        else:
            screen.bottom.background = unfocussed_colour
        screen.bottom.draw()
```

**Reply by asyncmind0** [reply]
Yea I tried something similar before not sure what I'm missing 

https://github.com/jagguli/qtile_config/blob/master/widgets/bar.py#L27

**Comment by hamensman** [comment]
Late to the party, but hopefully I can add something.

Check out ArcoLinux's qtile `config.py`: https://github.com/arcolinux/arcolinux-qtile/blob/master/etc/skel/.config/qtile/config.py

They declare the layouts (with the focus colours) like this:
```
layouts = [
    layout.MonadTall(margin=8, border_width=2, border_focus="#5e81ac", border_normal="#4c566a"),
    layout.MonadWide(margin=8, border_width=2, border_focus="#5e81ac", border_normal="#4c566a"),
    layout.Matrix(**layout_theme),
    layout.Bsp(**layout_theme),
    layout.Floating(**layout_theme),
    layout.RatioTile(**layout_theme),
    layout.Max(**layout_theme)
]
```
So here, you specify the colour of a window's border (in a given layout) when it is in focus, and when it is not in focus.

**Reply by asyncmind0** [reply]
Yes this works for window border, whats missing is for the screen Bar.

libqtile.bar.Bar class

**Reply by elParaguayo** [reply]
Did you try m-col's solution? I would have expected the hook to be the right approach here.

If it didn't work, can you explain what didn't work.
